omnisafe/evaluator.py

```diff
@@ -366,9 +386,17 @@ def evaluate(
                             'The policy must be provided or created before evaluating the agent.',
                         )
                 obs, rew, cost, terminated, truncated, _ = self._env.step(act)
+                if 'Saute' in self._cfgs['algo'] or 'Simmer' in self._cfgs['algo']:
+                    self._safety_obs -= cost.unsqueeze(-1) / self._safety_budget
+                    self._safety_obs /= self._cfgs.algo_cfgs.saute_gamma
 
                 ep_ret += rew.item()
                 ep_cost += (cost_criteria**length) * cost.item()
+                if (
```

Comment:
maybe we can consider wrapper env using `simmer` and `saute` adapter directly, It can simplify the process of evaluating and show the power of Adapter mechanism.

Reply:
Sounds great but, if do that, model-based evaluation must be rewritten. I think we can provide this feature in the future.

Reply:
I agree